pmd-core/src/main/java/net/sourceforge/pmd/AbstractPropertyDescriptorFactory.java

```diff
@@ -0,0 +1,321 @@
+ *
+ * @author Brian Remedios
+ */
+public abstract class AbstractPropertyDescriptorFactory<T> implements PropertyDescriptorFactory<T> {
+
+
+
+
+    static {
+        System.err.println(CORE_FIELD_TYPES_BY_KEY);
+    }
+
+
+    private final Class<?> valueType;
+
+    /**
+     * Denote the identifiers of the expected fields paired with booleans
+     * denoting whether they are required (non-null) or not.
+     */
+    private final Map<PropertyDescriptorField, Boolean> expectedFields;
+
+
+    public AbstractPropertyDescriptorFactory(Class<?> theValueType) {
+        valueType = theValueType;
+        expectedFields = CORE_FIELD_TYPES_BY_KEY;
+    }
+
+
+    public AbstractPropertyDescriptorFactory(Class<?> theValueType, Map<PropertyDescriptorField, Boolean> additionalFieldTypesByKey) {
+
+        valueType = theValueType;
+        if (additionalFieldTypesByKey == null) {
+            expectedFields = CORE_FIELD_TYPES_BY_KEY;
+            return;
+        }
+        Map<PropertyDescriptorField, Boolean> temp
+            = new HashMap<>(CORE_FIELD_TYPES_BY_KEY.size() + additionalFieldTypesByKey.size());
+        temp.putAll(CORE_FIELD_TYPES_BY_KEY);
+        temp.putAll(additionalFieldTypesByKey);
+
+        expectedFields = Collections.unmodifiableMap(temp);
+    }
+
+
+    @Override
+    public Class<?> valueType() {
+        return valueType;
+    }
+
+
+    @Override
+    public Set<PropertyDescriptorField> expectableFields() {
+        return Collections.unmodifiableSet(expectedFields.keySet());
+    }
+
+
+    /**
+     * Retrieves the name of the descriptor from the map.
+     *
+     * @param valuesById Map of attributes
+     *
+     * @return The name, which is null if none is specified
+     */
+    protected String nameIn(Map<PropertyDescriptorField, String> valuesById) {
+        return valuesById.get(NAME);
+    }
+
+
+    /**
+     * Retrieves the description from the map.
+     *
+     * @param valuesById Map of attributes
+     *
+     * @return The description, which is null if none is specified
+     */
+    protected String descriptionIn(Map<PropertyDescriptorField, String> valuesById) {
+        return valuesById.get(DESCRIPTION);
+    }
+
+
+    /**
+     * Retrieves the default value from the map.
+     *
+     * @param valuesById Map of attributes
+     *
+     * @return The default value
+     *
+     * @throws RuntimeException if the default value is null, empty, or missing
+     */
+    protected String defaultValueIn(Map<PropertyDescriptorField, String> valuesById) {
+        String deft = valuesById.get(DEFAULT_VALUE);
+        if (StringUtil.isEmpty(deft)) {
+            throw new RuntimeException("Default value was null, empty, or missing");
+        }
+        return deft;
+    }
+
+
+    @Override
+    public final PropertyDescriptor<T> createWith(Map<PropertyDescriptorField, String> valuesById) {
+      //  checkRequiredFields(valuesById);
```

Comment:
Right I commented it out to debug something. I'll push a commit to remove all traces of debugging